### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by NiranEC77** [vmware-tanzu/octant#385]
Hi
I am getting this error on an ubuntu server - panic: runtime error: invalid memory address or nil pointer dereference
followed: https://github.com/vmware-tanzu/octant/issues/132
I am using the latest build - https://github.com/vmware-tanzu/octant/releases/tag/v0.8.0

log output

github.com/vmware/octant/internal/dash.(*dash).Run
        /go/src/github.com/vmware/octant/internal/dash/dash.go:349
github.com/vmware/octant/internal/dash.Run.func1
        /go/src/github.com/vmware/octant/internal/dash/dash.go:190
2019-10-29T08:47:55.042-0700    ERROR   api/websocket_service.go:42     create websocket client {"err": "websocket: request origin not allowed by Upgrader.CheckOrigin"}
github.com/vmware/octant/internal/api.serveWebsocket
        /go/src/github.com/vmware/octant/internal/api/websocket_service.go:42
github.com/vmware/octant/internal/api.websocketService.func1
        /go/src/github.com/vmware/octant/internal/api/websocket_service.go:34
net/http.HandlerFunc.ServeHTTP
        /usr/local/go/src/net/http/server.go:1995
github.com/vmware/octant/internal/api.rebindHandler.func1.1
        /go/src/github.com/vmware/octant/internal/api/middleware.go:52
net/http.HandlerFunc.ServeHTTP
        /usr/local/go/src/net/http/server.go:1995
github.com/vmware/octant/vendor/github.com/gorilla/mux.(*Router).ServeHTTP
        /go/src/github.com/vmware/octant/vendor/github.com/gorilla/mux/mux.go:162
github.com/vmware/octant/vendor/github.com/gorilla/mux.(*Router).ServeHTTP
        /go/src/github.com/vmware/octant/vendor/github.com/gorilla/mux/mux.go:162
github.com/vmware/octant/vendor/github.com/gorilla/handlers.(*cors).ServeHTTP
        /go/src/github.com/vmware/octant/vendor/github.com/gorilla/handlers/cors.go:135
net/http.serverHandler.ServeHTTP
        /usr/local/go/src/net/http/server.go:2774
net/http.(*conn).serve
        /usr/local/go/src/net/http/server.go:1878
panic: runtime error: invalid memory address or nil pointer dereference
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x1378d95]

goroutine 179 [running]:
github.com/vmware/octant/internal/api.(*WebsocketClient).readPump.func1(0x0)
        /go/src/github.com/vmware/octant/internal/api/websocket_client.go:95 +0x5
panic(0x165e180, 0x32e47a0)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/vmware/octant/internal/api.(*WebsocketClient).readPump(0x0)
        /go/src/github.com/vmware/octant/internal/api/websocket_client.go:106 +0x72
created by github.com/vmware/octant/internal/api.serveWebsocket
        /go/src/github.com/vmware/octant/internal/api/websocket_service.go:46 +0xac

**Comment by wwitzel3** [comment]
Are you trying to connect to octant running on another host other than localhost?

**Comment by NiranEC77** [comment]
Yes I am trying to connect from another host
I ran the following command to start octant : OCTANT_LISTENER_ADDR=x.x.x.x:8900 octant
k8s version  v1.16.2 
Ubuntu 18.04.3 LTS

**Comment by NiranEC77** [comment]
> Are you trying to connect to octant running on another host other than localhost? Also what platform are you running on?

Yes I am trying to connect from another host
I ran the following command to start octant : OCTANT_LISTENER_ADDR=x.x.x.x:8900 octant
k8s version v1.16.2
Ubuntu 18.04.3 LTS

**Comment by GuessWhoSamFoo** [comment]
See if setting `OCTANT_ACCEPTED_HOSTS` helps

**Comment by wwitzel3** [comment]
As @GuessWhoSamFoo pointed out, setting `OCTANT_ACCEPTED_HOSTS` with the value of the host you are trying to connect to Octant from. You can also set the `OCTANT_LISTENER_ADDR` to  `0.0.0.0` which by default will allow all hosts

**Comment by wwitzel3** [comment]
@NiranEC77 any luck with setting the accepted hosts?

**Comment by moderation** [comment]
I'm getting the same panic and can reliably reproduce by clicking on the first entry in a section ie. Pods, Deployments etc. Running on RHEL 7 and compiled from source at master.
```
2019-11-01T16:51:29.747Z        DEBUG   api/websocket_state.go:171      setting content path    {"contentPath": "overview/namespace/REDACTED/workloads/deployments/envoy-dply"}
2019-11-01T16:51:30.484Z        DEBUG   api/content_manager.go:117      generating content      {"client-id": "d608f8fb-fcc7-11e9-bde2-548028d371d9", "contentPath": "overview/namespace/REDACTED/workloads/deployments/envoy-dply", "elapsed": "256.743784ms"}
2019-11-01T16:51:30.678Z        DEBUG   objectstore/dynamic_cache.go:90 informer cache has synced       {"key": "CacheKey[Namespace='REDACTED', APIVersion='extensions/v1beta1', Kind='ReplicaSet']", "elapsed": "200.220262ms"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x155e5e8]

goroutine 1261 [running]:
github.com/vmware-tanzu/octant/internal/queryer.(*ObjectStoreQueryer).ConfigMapsForPod(0xc001074040, 0x2560000, 0xc000a303c0, 0xc000990c00, 0x7, 0xc0003d2580, 0xb, 0x0, 0x0)
        /home/n044421/Library/octant/internal/queryer/queryer.go:746 +0x1d8
github.com/vmware-tanzu/octant/internal/objectvisitor.(*Pod).Visit.func3(0xc000d44768, 0x0)
        /home/n044421/Library/octant/internal/objectvisitor/pod.go:105 +0x90
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc000a30540, 0xc000be6c30)
        /home/n044421/Library/code/pkg/mod/golang.org/x/sync@v0.0.0-20190423024810-112230192c58/errgroup/errgroup.go:57 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
        /home/n044421/Library/code/pkg/mod/golang.org/x/sync@v0.0.0-20190423024810-112230192c58/errgroup/errgroup.go:54 +0x66
```

**Comment by wwitzel3** [comment]
@moderation the error you are getting is due to a nil ref error in one of our queryer methods. That is going to be fixed today (see #400). The error @NiranEC77 is getting is related to our origin check for websockets and is unrelated.
